File: 1441.cpp
struct Solution {
    vector<string> buildArray(const vector<int>& target, int n) {
        int nextNum = 0;
        vector<string> res;
        for (int n : target){
            while (true){
                ++nextNum;
                res.push_back("Push");
                if (nextNum != n){
                    res.push_back("Pop");
                } else {
                    break;
                }
            }
        }
        return res;
    }
};

File: sol/1441/1441.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        nextNum = 0
        res = list()
        for num in target:
            while True:
                nextNum += 1
                res.append("Push")
                if nextNum != num:
                    res.append("Pop")
                else:
                    break
        return res
